### ISSUE TITLE
Usage example

### ISSUE THREAD

**Issue by piranna** [dawsonjon/verilog-math#2]
The modules on your `fpu` repo has wires to signal when they are ready to accept more data or when the operations have finished, but on this one there's no such thing. How to know when an operation has finished? Inspecting the generated code there's a note on the header about how much cycles does they spend, do I need to count externally that cycles to know when they have finished? Is that duration warranted to be the same for every input?

**Comment by dawsonjon** [comment]
Yes the modules are all fully pipelined, they can accept a new input every
clock cycle, the output will appear N clock cycles later where N is the
latency listed in the module. This allows a throughput rate equal to the
clock rate, but you may find it difficult to take full advantage of this on
your application.

The modules all have different latencies, but they will be the same for
every calculation regardless of the input data.


On 26 Apr 2017 1:26 p.m., "Jesús Leganés-Combarro" <notifications@github.com>
wrote:

The modules on your fpu repo has wires to signal when they are ready to
accept more data or when the operations have finished, but on this one
there's no such thing. How to know when an operation has finished?
Inspecting the generated code there's a note on the header about how much
cycles does they spend, do I need to count externally that cycles to know
when they have finished? Is that duration warranted to be the same for
every input?

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/dawsonjon/verilog-math/issues/2>, or mute the thread

**Comment by piranna** [comment]
> the modules are all fully pipelined

Ok, now I understand why did you say these ones where better that the other repo... I though these ones where multi-stage too but in a sequential fashion and improvements where only done on the number of stages, not that this ones where pipelined instead, good job :-)


> you may find it difficult to take full advantage of this on
> your application

Yes, at this moment it's of not too much benefict since currently my CPU is sequential, but hope to convert it to be pipelined in the future.
